src/masternodes/balances.h

```diff
@@ -254,6 +254,108 @@ struct CTransferDomainMessage {
     }
 };
 
+struct CStatsBalances {
+    TAmounts balances;
+
+    Res Add(CTokenAmount amount) {
+        if (amount.nValue == 0) {
+            return Res::Ok();
+        }
+
+        balances[amount.nTokenId] += amount.nValue;
```

Comment:
We should use SafeAdd static function in amount.h to prevent value overflow

Reply:
Overflow errors yes, but we don't catch this anywhere. We change this later once we have the ability to propagate back throughout.